src/test/Microsoft.NET.HostModel.Tests/BundlerConsistencyTests.cs

```diff
@@ -0,0 +1,168 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using Xunit;
+using Microsoft.DotNet.CoreSetup.Test;
+using Microsoft.NET.HostModel.Bundle;
+
+namespace Microsoft.NET.HostModel.Tests
+{
+    public class BundlerConsistencyTests : IClassFixture<BundlerConsistencyTests.SharedTestState>
+    {
+        private SharedTestState sharedTestState;
+
+        public BundlerConsistencyTests(SharedTestState fixture)
+        {
+            sharedTestState = fixture;
+        }
+
+        private static string GetHostName(TestProjectFixture fixture)
+        {
+            return Path.GetFileName(fixture.TestProject.AppExe);
+        }
+
+        private static string GetPublishPath(TestProjectFixture fixture)
+        {
+            return Path.Combine(fixture.TestProject.ProjectDirectory, "publish");
+        }
+
+        private static DirectoryInfo GetBundleDir(TestProjectFixture fixture)
+        {
+            return Directory.CreateDirectory(Path.Combine(fixture.TestProject.ProjectDirectory, "bundle"));
+        }
+
+        [Fact]
+        public void TestWithEmptySpecFails()
+        {
+            var fixture = sharedTestState.TestFixture
+                .Copy();
+
+            var hostName = GetHostName(fixture);
+            var bundleDir = GetBundleDir(fixture);
+            Bundler bundler = new Bundler(hostName, bundleDir.FullName);
+
+            FileSpec[][] invalidSpecs =
+            {
+                new FileSpec[] {new FileSpec(hostName, null) },
+                new FileSpec[] {new FileSpec(hostName, "") },
+                new FileSpec[] {new FileSpec(hostName, "    ") }
+            };
+
+            foreach (var invalidSpec in invalidSpecs)
+            {
+                Assert.Throws<ArgumentException>(() => bundler.GenerateBundle(invalidSpec));
+            }
+        }
+
+        [Fact]
+        public void TestWithoutSpecifyingHostFails()
+        {
+            var fixture = sharedTestState.TestFixture
+                .Copy();
+
+            var hostName = GetHostName(fixture);
+            var appName = Path.GetFileNameWithoutExtension(hostName);
```

Comment:
Should these tests also be using a `BundleApp` helper?

Reply:
The `BundleApp()` helper is a temporary step until Core-setup is updated to use 3.0 SDK for its testing. https://github.com/dotnet/core-setup/issues/5337 
Once the new SDK is available, `BundleApp` helper should be removed in favor of passing the `/p:PublishSingleFile=true` property to publish.

However this test should continue to invoke the bundler directly, since it is a test for the `HostModel` library.